### ISSUE TITLE
Feature request: new object upper-sexp

### ISSUE THREAD

**Issue by s-kostyaev** [paldepind/composable.el#17]
Firstly thanks for this great package. I think that new text object upper-sexp with behavior like expand-region will be very useful. I don't see package internals yet, but maybe I'll try to add this object and send PR if you agree to add it and don't want to do it yourself.

**Comment by paldepind** [comment]
Late response. Sorrry :cry: 

I actually don't know what `upper-sexp` is? Is it a new form of object?

**Comment by s-kostyaev** [comment]
Did you try expand-region? If you inside deep nested sexp it will firstly mark nested sexp then previous level sexp. It's like going by the tree step by step from leaf to the root. I think that text object with same behavior will be very useful. If you don't try expand-region yet you should try it.

**Comment by s-kostyaev** [comment]
In this object begin of the region must be set with `(backward-up-list)` and end of region must be set with `(up-list)`.

**Comment by s-kostyaev** [comment]
See the PR.

**Comment by paldepind** [comment]
Thank you @s-kostyaev.

**Comment by s-kostyaev** [comment]
@paldepind If I use up-list with composable-mark-mode all works fine, but if I use it with command (composable-kill-buffer for example) it works like I press it triple times. I think that "C-w o" must be equivalent with "C-SPC  o C-w" but first time it works like "C-SPC ooo C-w". But composing works like single command:
 "C-w o" -> "C-SPC ooo C-w"
 "C-w oo" -> "C-SPC oooo C-w"
 "C-w ooo" -> "C-SPC ooooo C-w"
etc.

TL/DR: test case
```
  Scenario: Kill list from inside
    When I insert "(deep (nested (list arg1 arg2)))"
    And I place the cursor after "1"
    And I press "C-w o"
    Then I should see "(deep (nested ))"
```
will fail for now. Maybe you know how can I debug it?
